### ISSUE TITLE
Cursor error when using MongoMock

### ISSUE THREAD

**Issue by defrank** [Scille/umongo#117]
When using `Document.find`, the MongoMock cursor does not have `Cursor.__id`.  The problem seems to be that `WrappedCursor` inherits from `pymongo.cursor.Cursor`, but `WrappedCursor.raw_cursor` is an instance of `mongomock.collection.Cursor`, which does not set `__id`.

```python
from mongomock import MongoClient
from umongo import Instance, Document


client = MongoClient()
db = client.test
instance = Instance(db)


@instance.register
class Something(Document):
    """Just care about `Something.find`."""


if __name__ == '__main__':
    Something().commit()
    list(Something.find())
```

```
Exception ignored in: <bound method Cursor.__del__ of <umongo.frameworks.pymongo.WrappedCursor object at 0x7f87382aeca8>>
Traceback (most recent call last):
  File "/home/derek/Code/fanai-system/venv/lib/python3.6/site-packages/pymongo/cursor.py", line 225, in __del__
    self.__die()
  File "/home/derek/Code/fanai-system/venv/lib/python3.6/site-packages/pymongo/cursor.py", line 277, in __die
    if self.__id and not self.__killed:
  File "/home/derek/Code/fanai-system/venv/lib/python3.6/site-packages/umongo/frameworks/pymongo.py", line 28, in __getattr__
    return getattr(self.raw_cursor, name)
AttributeError: 'Cursor' object has no attribute '_Cursor__id'
```

**Comment by touilleMan** [comment]
Can you try with the mongomock codebase from [this PR](https://github.com/mongomock/mongomock/pull/325) ? I've ironed a lot of corner cases with mongomock, but there are quiet slow to merge it...

**Comment by defrank** [comment]
Still didn't work, @touilleMan.  

What I did to test:
```sh
git clone git@github.com:touilleMan/mongomock.git
cd mongomock
git co -b multiple-corrections origin/multiple-corrections

python3 -m venv venv
source venv/bin/activate
pip install umongo pymongo mongomock

PYTHONPATH=$PWD python3 test.py  # to override mongomock
```

test.py:
```python
from mongomock import MongoClient
from umongo import Instance, Document


client = MongoClient()
db = client.test
instance = Instance(db)


@instance.register
class Something(Document):
    """Just care about `Something.find`."""


if __name__ == '__main__':
    Something().commit()
    list(Something.find())
```

Output:
```sh
(venv) % PYTHONPATH=$PWD python3 test.py                                                                                                                                                                                      
Exception ignored in: <bound method Cursor.__del__ of <umongo.frameworks.pymongo.WrappedCursor object at 0x10190f630>>
Traceback (most recent call last):
  File "/Users/dmf/Code/mongomock/venv/lib/python3.6/site-packages/pymongo/cursor.py", line 225, in __del__
    self.__die()
  File "/Users/dmf/Code/mongomock/venv/lib/python3.6/site-packages/pymongo/cursor.py", line 277, in __die
    if self.__id and not self.__killed:
  File "/Users/dmf/Code/mongomock/venv/lib/python3.6/site-packages/umongo/frameworks/pymongo.py", line 28, in __getattr__
    return getattr(self.raw_cursor, name)
AttributeError: 'Cursor' object has no attribute '_Cursor__id'
```

**Comment by touilleMan** [comment]
I've fixed the bug here 91e377859a790c2dbbba10e9ecf4ccec4d626b21 ^^

**Comment by defrank** [comment]
I confirm fixed by 91e377859a790c2dbbba10e9ecf4ccec4d626b21
